clientv3/integration/kv_test.go

```diff
@@ -251,3 +253,45 @@ func TestKVDelete(t *testing.T) {
 		t.Fatalf("gresp.Kvs got %+v, want none", gresp.Kvs)
 	}
 }
+
+func TestKVCompact(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	kv := clientv3.NewKV(clus.RandClient())
+
+	for i := 0; i < 10; i++ {
+		if _, err := kv.Put("foo", "bar", lease.NoLease); err != nil {
+			t.Fatalf("couldn't put 'foo' (%v)", err)
+		}
+	}
+
+	err := kv.Compact(7)
+	if err != nil {
+		t.Fatalf("couldn't compact kv space (%v)", err)
+	}
+	err = kv.Compact(7)
+	if err == nil || err != v3rpc.ErrCompacted {
+		t.Fatalf("error got %v, want %v", err, v3rpc.ErrFutureRev)
+	}
+
+	wc := clientv3.NewWatcher(clus.RandClient())
+	defer wc.Close()
+	wchan := wc.Watch(context.TODO(), "foo", 3)
+
+	wr, ok := <-wchan
+	isEmpty := reflect.DeepEqual(clientv3.WatchResponse{Header: pb.ResponseHeader{}, Events: nil}, wr)
```

Comment:
remove the empty checking since this ensures by go runtime that when a chan is closed it returns empty response.